### PR TITLE
Add tests for template handling

### PR DESCRIPTION
**Pull request by twaugh**
Signed-off-by: Tim Waugh <twaugh@redhat.com>

### CODE REVIEW

**Review by mprahl — APPROVED** [review]

File: main_test.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		},
+		{
+			"compatibility",
+			[]string{"retrodep", "-template", "@{{.Rev}}", "."},
+			"{{.Pkg}}@{{.Rev}}",
+		},
+	}
+
+	for _, tc := range tcs {
+		tc := tc
```
Comment (vkrizan) [inline main_test.go:44]: What effect does this re-assignment have?

Comment (twaugh) [reply main_test.go:44]: The `tc` used by the for loop gets a different value each time round the loop. The `t.Run` call runs the provided function in its own goroutine, so `tc` in that function comes from the closure for running it in. With `tc := tc` we create a new variable for each new goroutine we create, ensuring they don't all run for the same `tc`.

More information: https://golang.org/doc/faq#closures_and_goroutines

Comment (alexandrevicenzi) [reply main_test.go:44]: Can you add a comment to this code? Unaware people might remove this. :grimacing:

Comment (vkrizan) [reply main_test.go:44]: Got it. Thanks @twaugh

**Review by vkrizan — APPROVED** [review]

**Comment by RalphCorderoy** [comment]
Hi @alexandrevicenzi, I don't think it needs `foo := foo` needs a comment. _it's legal and idiomatic in Go to do this_ ― https://golang.org/doc/effective_go.html
I realise if the idiom is unknown then it's a bit of a puzzle, but every language has its quirks that have to be learnt. Another is Go's _blank identifier_, the underscore. https://golang.org/doc/effective_go.html#blank
It may look like it does in Python, but in Go it's a defined part of the language whereas in Python it's just another identifier.

**Review by litleleprikon — APPROVED** [review]

**Pull request merged**